### ISSUE TITLE
Pasting in a string that is 1043 characters during interactive text input causes repeats in prompt

### ISSUE THREAD

**Issue by eyevanovich** [pterm/pterm#560]
When trying to input a very long string of characters, in this case a long JWT token. I'm getting repeated command prompts that show up. The good thing is that it allows me to get over the 1024 byte buffer limitation in the terminal but it's not very appealing to look at. Is it possible to fix this?

![kitty-2023-08-21 at 11 01 41](https://github.com/pterm/pterm/assets/54329731/8c765c86-199f-4f36-99ee-6d630e729d5a)

**Comment by MarvinJWendt** [comment]
Might be the same problem as in #557.
Thanks for reporting! We'll take look.

**Comment by floaust** [comment]
I will take a look at it tomorrow. But it seems like our logic does not handle it well atm when the input is larger than one line. @MarvinJWendt correct me if I am wrong

**Comment by kpurdon** [comment]
I was able to workaround this (short-term, not-ideal) by trim/padding each string going in to max terminal width - a buffer, so ya it seems like the root issue is pterm isn't handling text > terminal width.

Staying tuned, thanks for the response!

**Comment by eyevanovich** [comment]
Pretty sure text larger than the terminal width is the issue because if you try to resize the terminal with the text shown in the screenshot above, it will redraw every line again and again without any additional input.

**Comment by MarvinJWendt** [comment]
Yes, the issue is, that we move the cursor around the terminal on key input. This is mainly done for the `TextArea` (`TextInput.WithMultiline()`) as we need to move the cursor up, down, left and right. When the string goes over the terminals width, the cursor jumps down automatically, which introduces an offset on our calculated position, resulting in duplicate lines.

**Comment by kpurdon** [comment]
Hello! Any progress on this issue?

**Comment by emmahsax** [comment]
I'm seeing this issue too. Like the others, it's mostly a cosmetic issue.

**Comment by Homoarea** [comment]
I’ve found a solution.

Root cause: Strings that exceed the terminal’s width are subject to soft line breaks (i.e. automatic line breaks), causing the `area` position to shift downwards. When `area` is refreshed again, the internally called `Clear()` method fails to clear the text segments left over due to misalignment; consequently, every call to `area.Update(content)` results in repeated printing.

Implementation approach: Before calling `area.Update(content)`, convert the `content` string to a format that fits the terminal’s width, thereby preventing the terminal from inserting soft line breaks.

This resolves the issue of repeated printing caused by input exceeding the terminal’s width or the `content` string exceeding the terminal’s width; however, it does not address repeated printing caused by input exceeding the terminal’s height, nor the occasional repetition resulting from `area` misalignment due to soft line breaks triggered by changes to the terminal’s width.

I will submit a PR shortly, linking to this issue. Incidentally, #557 and #714 are duplicates of this one.
